### ISSUE TITLE
ci: docker build failed with `detach` not found

### ISSUE THREAD

**Issue by skyzh** [singularity-data/risingwave#2782]
```
#19 174.6 error[E0599]: no method named `detach` found for struct `tokio::task::JoinHandle` in the current scope
1957
#19 174.6    --> src/stream/src/task/stream_manager.rs:569:26
1958
#19 174.6     |
1959
#19 174.6 569 |                         .detach();
1960
#19 174.6     |                          ^^^^^^ method not found in `tokio::task::JoinHandle<()>`
1961
#19 174.6 
1962
#19 174.9 For more information about this error, try `rustc --explain E0599`.
1963
#19 174.9 error: could not compile `risingwave_stream` due to previous error
1964
#19 174.9 warning: build failed, waiting for other jobs to finish...
```

I guess this is somehow related to madsim? cc @wangrunji0408
